src/message_thread.cc

```diff
@@ -170,30 +173,35 @@ namespace Astroid {
       return;
 
     } else {
-      GError *err = NULL; (void) (err); // not used in GMime 2.
-      GMimeStream   * stream  = g_mime_stream_file_open (fname.c_str(), "r", &err);
-      g_mime_stream_file_set_owner (GMIME_STREAM_FILE(stream), TRUE);
+      GMimeStream * stream = NULL;
+# ifndef DISABLE_PLUGINS
+      stream = astroid->plugin_manager->astroid_extension->process (fname.c_str());
+# endif
       if (stream == NULL) {
-        LOG (error) << "failed to open file: " << fname << " (unspecified error)";
+        GError *err = NULL; (void) (err); // not used in GMime 2.
+        stream  = g_mime_stream_file_open (fname.c_str(), "r", &err);
+        g_mime_stream_file_set_owner (GMIME_STREAM_FILE(stream), TRUE);
+        if (stream == NULL) {
+          LOG (error) << "failed to open file: " << fname << " (unspecified error)";
 
-        has_file = false;
-        missing_content = true;
+          has_file = false;
+          missing_content = true;
 
-        if (in_notmuch) {
-          LOG (warn) << "loading cache for missing file from notmuch";
-          load_notmuch_cache ();
-        } else {
-          LOG (error) << "tried to open disk file, but failed, message is not in database either.";
-          string error_s = "failed to open file: " + fname;
-          throw message_error (error_s.c_str());
+          if (in_notmuch) {
```

Comment:
The cache loading from notmuch now depends on what the plugin does and wether it manages to open the file. What happens if there's an error in parsing the file? Or if the file does not exist?

Reply:
In either case the plugin should return a NULL pointer (unless it's doing some reading from a cache itself) and go into this branch of the code. Technically this will try to open the same file again and fail again which isn't necessary, but it keeps the code simpler.